br/pkg/stream/rewrite_meta_rawkv.go

```diff
@@ -95,164 +98,215 @@ func NewSchemasReplace(
 	}
 }
 
-func (sr *SchemasReplace) rewriteKeyForDB(key []byte, cf string) ([]byte, bool, int64, error) {
+// TidyOldSchemas produces schemas information.
+func (sr *SchemasReplace) TidyOldSchemas() *backup.Schemas {
```

Comment:
shall we need  the old full backup schemas here, not only incremental backup schemas ?

Reply:
Yes, it need save all of upstream-cluster schemas(contains old snapshot schemas and incremental schemas) to the new backupmeta(filename= backupmeta_<restore-ts1>) for the next restore log. 
Because It need to rewrite the old snapshot schemas at the next restore log (from restore-ts1 to restore-ts2).